src/wallet/scriptpubkeyman.cpp

```diff
@@ -384,7 +384,7 @@ bool LegacyScriptPubKeyMan::Upgrade(int prev_version, std::string& error)
         hd_upgrade = true;
     }
     // Upgrade to HD chain split if necessary
-    if (m_storage.CanSupportFeature(FEATURE_HD_SPLIT)) {
+    if (m_storage.CanSupportFeature(FEATURE_HD_SPLIT) && CHDChain::VERSION_HD_CHAIN_SPLIT) {
```

Comment:
In commit "HD Split: Avoid redundant upgrades" (415afcccd3e5583defdb76e3a280f48e98983301):

This seems buggy and the change looks like it has no effect. Is this supposed to say `&& hdChain.nVersion < CHDChain::VERSION_HD_CHAIN_SPLIT`?